kafka_consumer/datadog_checks/kafka_consumer/client/confluent_kafka_client.py

```diff
@@ -35,7 +35,15 @@ def reset_offsets(self):
         raise NotImplementedError
 
     def get_partitions_for_topic(self, topic):
-        raise NotImplementedError
+        try:
+            cluster_metadata = self.kafka_client.list_topics(topic)
+            topic_metadata = cluster_metadata.topics[topic]
+            partitions = list(topic_metadata.partitions.keys())
+            return partitions
+        except KafkaException as e:
+            self.log.error("Received exception when getting partitions for topic %s: %s" % topic, e)
+            return None
```

Comment:
To reviewer: I am returning `None` here since the `kafka-python` [implementation](https://github.com/dpkp/kafka-python/blob/7ac6c6e29099ccba4d50f5b842972dd7332d0e58/kafka/cluster.py#L106-L117) of `get_partitions_for_topic()` returns this in the worst case.